src/libs/markAllPolicyReportsAsRead.ts

```diff
@@ -0,0 +1,33 @@
+// eslint-disable-next-line you-dont-need-lodash-underscore/each
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+import * as ReportActionFile from './actions/Report';
+import * as ReportUtils from './ReportUtils';
+
+export default function markAllPolicyReportsAsRead(policyID: string) {
+    const connectionID = Onyx.connect({
+        key: ONYXKEYS.COLLECTION.REPORT,
+        waitForCollectionCallback: true,
+        callback: (allReports) => {
+            if (!allReports) {
+                return;
+            }
+
+            let delay = 0;
+            Object.keys(allReports).forEach((key: string) => {
+                const report: Report | null | undefined = allReports[key];
+                if (report?.policyID !== policyID || !ReportUtils.isUnread(report)) {
+                    return;
+                }
+
+                setTimeout(() => {
+                    ReportActionFile.readNewestAction(report?.reportID);
+                }, delay);
+
+                delay += 1000;
```

Comment:
👋 @marcaaron or @rlinoz Do you recall why we are using a timeout here? We are looking to add similar functionality to a [keyboard shortcut](https://github.com/Expensify/App/pull/61753) and wanted to reuse same logic but the timeout is a problem as it will delay even optimistic actions.

Is this done to prevent server flooding or something else?

Reply:
Also, here is a slack thread for moving this logic to BE instead https://expensify.slack.com/archives/C01GTK53T8Q/p1746861626079909

Reply:
Yes, it's just to avoid hammering the server with requests since some users who requested this feature had thousands of reports. Though, I think we probably have solve that problem by now.